src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,134 @@
+package duke.parser;
+
+import duke.dukeexception.DukeKeywordMissingException;
+import duke.dukeexception.DukeTaskNonExistException;
+import duke.dukeexception.DukeUnknownInputException;
+import duke.dukeexception.EmptyDescriptionException;
+import duke.storage.Storage;
+import duke.tasklist.TaskList;
+import duke.ui.UI;
+
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+/**
+ * This class deals with the interpretation of the user's inputs by the program.
+ */
+public class Parser {
+    private final UI ui;
+
+    /**
+     * Constructor for the parser.
+     *
+     * @param ui UI which deals with interactions with the user
+     */
+    public Parser(UI ui) {
+        this.ui = ui;
+    }
+
+    /**
+     * Method that is used as a flag in the duke handler method in the duke class to indicate when the program should
+     * stop running.
+     *
+     * @param response Response would refer to the input by the user
+     * @return boolean value, true = the program should stop, false = the program should continue running
+     */
+    public boolean toStop(String response) {
+        return response.equals("bye");
+    }
+
+    /**
+     * Method that takes in different commands by the user and calls the relevant methods in the UI to deal with them.
+     *
+     * <p>
+     * <h3> Userinput = bye</h3>
+     * Stops the entire program
+     * </p>
+     *
+     * <p>
+     * <h3> UserInput = list</h3>
+     * Prints out a list of the tasks in the TaskList
+     * </p>
+     *
+     * <p>
+     * <h3> UserInput = find xxx task</h3>
+     * Finds the tasks that match the keyword provided by the user and prints it out as a list
+     * </p>
+     *
+     * <p>
+     * <h3> UserInput = delete xxx task</h3>
+     * Triggers the UI to delete the task and converts the one-based index given by the user
+     * to a zero-based index.
+     * </p>
+     *
+     * <p>
+     * <h3> UserInput = done</h3>
+     * Triggers the UI to mark the task as done and converts the one-based index given by the user
+     * to a zero-based index.
+     * </p>
+     *
+     * <p>
+     * <h3> UserInput = create Event/Deadline/Todo </h3>
+     * Triggers the UI to create the corresponding task object and separates the user's input to
+     * show the name of the task, type of task and the duedate of the task(only applicable for deadline and events).
+     * It may throw an error if the description is empty, keywords are missing or the dates are in the wrong format.
+     * </p>
+     *
+     * @param response String response which represents the user's input
+     */
+    public void listen(String response) {
+        try {
+            int indexer;
```

Comment:
Perhaps a more intuitive variable name here? If I understand correctly, it is used to determine the index of task in your tasklist, so perhaps `taskIndex` ?